src/effektManager.ts

```diff
@@ -0,0 +1,522 @@
+        }
+
+        this.updateStatusBar();
+    }
+
+    /**
+     * Checks for Effekt updates and offers to install/update if necessary.
+     * @returns A promise that resolves with the current Effekt version.
+     */
+    public async checkForUpdatesAndInstall(): Promise<string> {
+        try {
+            const effektPath = await this.locateEffektExecutable();
+            if (!this.effektVersion) {
+                const currentVersion = await this.fetchEffektVersion(effektPath.path);
+                this.effektVersion = currentVersion
+            }
+
+            const latestVersion = await this.getLatestNPMVersion(this.effektNPMPackage);
+
+            // check if the latest version strictly newer than the current version
+            if (!this.effektVersion || compareVersion(latestVersion, this.effektVersion, '>')) {
+                return this.promptForAction(latestVersion, 'update');
+            }
+
+            this.updateStatusBar();
+            return this.effektVersion || '';
+        } catch (error) {
+            if (error instanceof Error && error.message.includes('Effekt executable not found')) {
+                const latestVersion = await this.getLatestNPMVersion(this.effektNPMPackage)
+                return this.promptForAction(latestVersion, 'install');
+            } else {
+                this.showErrorWithLogs(`Failed to check Effekt: ${error}`);
+                return '';
+            }
+        }
+    }
+
+    /**
+     * Prompts the user to perform an action (install or update) and executes it if confirmed.
+     * @param version The version to install or update to.
+     * @param action The action to perform ('install' or 'update').
+     * @returns A promise that resolves with the installed/updated version or an empty string.
+     */
+    private async promptForAction(version: string, action: 'install' | 'update'): Promise<string> {
+        const message = action === 'update'
+            ? `A new version of Effekt is available (${version}). Would you like to update?`
+            : `Effekt ${version} is available. Would you like to install it?`;
+
+        const response = await vscode.window.showInformationMessage(message, 'Yes', 'No');
+        if (response === 'Yes') {
+            return this.installOrUpdateEffekt(version, action);
+        }
+        this.updateStatusBar();
+        return this.effektVersion || '';
+    }
+
+    /**
+     * Checks if Node.js and npm are installed and meet the minimum version requirements.
+     * @returns A promise that resolves with a boolean indicating if the requirements are met.
+     */
+    private async checkNodeAndNpm(): Promise<boolean> {
+        try {
+            const nodeVersion = await this.execCommand('node --version');
+            await this.execCommand('npm --version'); // Note: if needed, we could also check npm version.
+
+            const minNodeVersion = 'v12.0.0'; // Minimum supported Node.js version
```

Comment:
```suggestion
            const minNodeVersion = 'v16.0.0'; // Minimum supported Node.js version
```
Since https://github.com/effekt-lang/effekt/pull/599.